client/my-sites/themes/search-themes-tracks.js

```diff
@@ -0,0 +1,109 @@
+import { usePrevious } from '@wordpress/compose';
+import { useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { isRequestingThemesForQuery, prependThemeFilterKeys } from 'calypso/state/themes/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+function useThemeSearchDetails( query ) {
+	const search = query.search;
+	const searchTaxonomies = useSelector( ( state ) =>
+		prependThemeFilterKeys( state, query.filter )
+	);
+	const searchTerm = searchTaxonomies + search;
+	const prevSearchTerm = usePrevious( searchTerm );
+	const searchTermHasChanged = prevSearchTerm !== searchTerm && !! searchTerm && searchTerm !== '';
```

Comment:
The expectation is to pick all changes, except from searching _something_ to _empty_, right?

Reply:
Yes.
My reasoning is that if the search is empty, it's not really technically a search, is it?

It's basically the default unfiltered view. Logging this doesn't give us any particularly interesting info, although the more I think about it, the weaker my opinion is.

I guess I'm gonna keep excluding empty searches for now; we can revisit this decision in the future if we determine we need to know more about them too.

Reply:
I tend to agree with you, thanks for the explanation.